src/notice.ts

```diff
@@ -74,4 +77,24 @@ export function validateNotice(notice: Notice): void {
   if (!semver.validRange(notice.schemaVersion)) {
     throw new Error(`Schema version ${notice.schemaVersion} is not a valid semver range`);
   }
+}
+
+function validNoticeSemver(version: string) {
```

Comment:
I don't really like what we're doing here, it seems very purpose-built for this specific error case.

Can we not say something like: "if any component encodes a range, the entire expression must match more than one version" or something? That seems like a reasonable rule we're trying to encode?

Reply:
Tricky, because for that we need to know the available versions (or fabricate them with some).

Semver syntax supports _intersection_ of constraints: `>1 <3` means the intersection of all values that are greater than `1` and lower than `3`, i.e. `2`. I think what this PR is trying to do to explicitly disallow any intersections with single explicit version, as they can be written in a much simpler way and are likely a mistake.

Reply:
Alright, I went with an algorithmic check. Now it's using more built-in semver functionality instead of manual string parsing.